lib/puppet/provider/service/init.rb

```diff
@@ -152,6 +152,13 @@ def restartcmd
     (@resource[:hasrestart] == :true) && [initscript, :restart]
   end
 
+  def texecute(type, command, fof = true, squelch = false, combine = true)
+    if type =~ /^start/ && Facter.value(:osfamily) == "Solaris"
```

Comment:
Ideally this logic would live in a Solaris specific provider that inherited from `init`. But it appears the `init` provider has lots of platform specific tweaks already so probably ok as is.